torch/distributed/distributed_c10d.py

```diff
@@ -247,6 +247,7 @@ def register_backend(cls, name, func, extended_api=False, devices: Optional[Unio
         )
 
         setattr(Backend, name.upper(), name.lower())
+        setattr(torch.distributed, f'is_{name.lower()}_available', lambda: True)
```

Comment:
We expose `is_backend_available` as a public API:

https://github.com/pytorch/pytorch/blob/d455c33ccaf0fef467f7bd85880340a6b303e880/torch/distributed/distributed_c10d.py#L936-L953

I believe that should be sufficient?

Reply:
Yes, I found that `is_backend_available` can be used to check whether custom backent is available, but the way this interface is different from other backends. I want to make the user experience the same.

Reply:
I didn't say we are removing is_nccl_available, just that I wish we could have done things differently. 

If we did go through a deprecation cycle for it, it'd still take a long time before you'd be able to have it gone from most code, and it's work to do that. Is it worth it?